### PR TITLE
Jkmarx/groups api v2

### PR DESCRIPTION
**Pull request by jkmarx**
Ref #2618
- Add a group API V2 which supports query by dataSetUuid (will extend in the next few pull requests)
- Extend data set API V2 to return owner object and user_perms (previously sent by share api)
- ? Consider removing is_owner field in the future
- Update client to use the groups api to get groups with data set's perms
   - Had to reorganize client and rename components to better fit new api calls
- Add unit tests

**Can't remove data set resource until we extend group api v2 to include patch for updating perms.**

### CODE REVIEW

File: refinery/ui/source/js/data-set-about/ctrls/details-ctrl.js [hunk 1]
```diff
@@ -1,142 +1,145 @@
-'use strict';
+/**
```
Comment (jkmarx) [inline refinery/ui/source/js/data-set-about/ctrls/details-ctrl.js]: reorganized to meet style guide and use new api

File: refinery/ui/source/js/data-set-about/ctrls/groups-perms-ctrl.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/**
```
Comment (jkmarx) [inline refinery/ui/source/js/data-set-about/ctrls/groups-perms-ctrl.js]: Switched over the share_component to group_perms_component name, due to new apis.

File: refinery/ui/source/js/data-set-about/partials/details.html [hunk 3]
```diff
@@ -468,7 +468,33 @@ <h3>Data Files</h3>
   </span>
 </p>
 
-<rp-data-set-about-sharing></rp-data-set-about-sharing>
+<div class="refinery-header">
```
Comment (jkmarx) [inline refinery/ui/source/js/data-set-about/partials/details.html]: Now returned from data_set v2 api so moved into details component

File: refinery/core/test_views.py [hunk 4]
```diff
@@ -816,6 +847,91 @@ def test_is_owner_reflects_actual_owner_with_admin_requester(self):
...
+        )
+        self.data_set = create_dataset_with_necessary_models(user=self.user)
+        self.group = ExtendedGroup.objects.create(name="Test Group")
+        self.group_2 = ExtendedGroup.objects.create(name="Test Group 2")
+        self.group.user_set.add(self.user)
+        self.group_2.user_set.add(self.user)
+        self.data_set.share(self.group)
+        self.data_set.share(self.group_2)
+
+    def test_get_groups_with_ds_uuid_returns_401_for_anon(self):
```
Comment (hackdna) [inline refinery/core/test_views.py]: I'd expand `ds` to `dataset` to make it more explicit (here and below).

File: refinery/core/test_views.py [hunk 5]
```diff
@@ -816,6 +847,91 @@ def test_is_owner_reflects_actual_owner_with_admin_requester(self):
         self.assertFalse(data_set["is_owner"])
 
 
+class GroupApiV2Tests(APIV2TestCase):
+    def setUp(self):
+        super(GroupApiV2Tests, self).setUp(
+            api_base_name="groups/",
+            view=GroupViewSet.as_view({'get': 'list'})
```
Comment (hackdna) [inline refinery/core/test_views.py]: These two could fit on one line?

**Pull request merged**